src/shogun/classifier/svm/OnlineSVMSGD.cpp

```diff
@@ -145,7 +145,7 @@ bool OnlineSVMSGD::train(std::shared_ptr<Features> data)
 	float64_t wnorm = linalg::dot(m_w, m_w);
 	io::info("Norm: {:.6f}, Bias: {:.6f}", wnorm, bias);
 
-	return true;
+	return shared_from_this()->as<Machine>();
```

Comment:
this could be std::static_pointer_cast, since we know that `this` is a derived from `Machine`